### PR TITLE
lib: mpsl: Add the MPSL library

### PR DESCRIPTION
**Pull request by sofu1**
Add the obfuscated library files, and the public header files.
Add references to MPSL lib and folders to CMakeLists and Kconfig
files so that MPSL can be linked in.
For now hardcoded to only link the soft version in mpsl/CMakeLists.

Signed-off-by: Solveig Fure <solveig.fure@nordicsemi.no>

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -7,6 +7,7 @@
 set(NRFXLIB_DIR ${CMAKE_CURRENT_LIST_DIR} CACHE PATH "NRFXLIB root directory")
 add_subdirectory_ifdef(CONFIG_NRFXLIB_NFC             nfc)
 add_subdirectory_ifdef(CONFIG_BT_LL_NRFXLIB           ble_controller)
+add_subdirectory_ifdef(CONFIG_MPSL_NRFXLIB            mpsl)
```
Comment (rugeGerritsen) [inline CMakeLists.txt]: There will not be another MPSL. Maybe we should name it `MPSL_LIBRARY` ?

File: mpsl/README.rst [hunk 2]
```diff
@@ -0,0 +1,21 @@
+.. _mpsl:
+
+MPSL
+====
+
+Multi-Protocol Service Layer
+
```
Comment (rugeGerritsen) [inline mpsl/README.rst:7]: Add some introduction text here describing the main purpose and functionality of the library?

File: mpsl/doc/mpsl.rst [hunk 3]
```diff
@@ -0,0 +1,6 @@
+.. _mpsl_readme:
+
+How to use
+==========
+
+TODO
```
Comment (rugeGerritsen) [inline mpsl/doc/mpsl.rst]: TODO

File: mpsl/README.rst (hunk 2 above)
Comment (sofu1) [reply mpsl/README.rst:7]: Can add something like this for now: "Multi-Protocol Service Layer (MPSL) is a library of common services for single and multi-protocol wireless implementations."
But we should add more stuff here later.

File: mpsl/CMakeLists.txt [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+nrfxlib_calculate_lib_path(lib_path)
+
+set(MPSL_LIB_PATH ${CMAKE_CURRENT_SOURCE_DIR}/${lib_path})
+
+if(NOT EXISTS ${MPSL_LIB_PATH})
+  message(WARNING "This combination of SoC and floating point ABI is not supported by the MPSL lib."
+                  "(${MPSL_LIB_PATH} doesn't exist.)")
+endif()
+
+set(MPSL_LIB ${MPSL_LIB_PATH}/libmpsl_soft_nrf52.a)
```
Comment (sofu1) [inline mpsl/CMakeLists.txt]: remove?

File: mpsl/Kconfig [hunk 5]
```diff
@@ -0,0 +1,12 @@
+#
+# Copyright (c) 2018 Nordic Semiconductor
```
Comment (sofu1) [inline mpsl/Kconfig]: 2019

File: mpsl/doc/mpsl.rst (hunk 3 above)
Comment (sofu1) [reply mpsl/doc/mpsl.rst]: I'll remove this

File: CMakeLists.txt (hunk 1 above)
Comment (sofu1) [reply CMakeLists.txt]: Done

File: mpsl/README.rst (hunk 2 above)
Comment (sofu1) [reply mpsl/README.rst:7]: Done

File: mpsl/doc/mpsl.rst (hunk 3 above)
Comment (sofu1) [reply mpsl/doc/mpsl.rst]: Added a brief explanation

File: mpsl/CMakeLists.txt (hunk 4 above)
Comment (sofu1) [reply mpsl/CMakeLists.txt]: Removed the set(..)

File: mpsl/Kconfig (hunk 5 above)
Comment (sofu1) [reply mpsl/Kconfig]: Fixed

**Pull request closed without merging**